### PR TITLE
feat(FEC-10290): upgrade NPM packages

### PR DESCRIPTION
**Pull request by dan-ziv**
### Description of the Changes

Please add a detailed description of the change, weather it's an enhancement or a bugfix.
If the PR is related to an open issue please link to it.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -93,5 +93,8 @@ module.exports = {
       amd: 'kaltura-player-js',
       root: ['KalturaPlayer']
     }
+  },
+  optimization: {
+    minimize: PROD
```
Comment (OrenMe) [inline webpack.config.js:98]: use `--mode production`/`--mode development` in package.json instead(for build/dev/watch scripts)

File: dist/playkit-cast-receiver.js [hunk 2]
```diff
@@ -1,2 +1,2 @@
...n t._player.removeEventListener(e,t._playerEventHandlers[e])}),this._adProgressIntervalId&&(clearInterval(this._adProgressIntervalId),this._adProgressIntervalId=null))}},{key:"_getAdBreakOptions",value:function(e){var t={},a=this._playerManager.getBreakManager().getBreakById(e.breakId);return a&&(t.position=a.position,t.type=this._getAdBreakTypeByPosition(a.position),t.numAds=a.breakClipIds.length),t}},{key:"_getAdBreakTypeByPosition",value:function(e){switch(e){case 0:return d.PRE;case-1:return d.POST;default:return d.MID}}},{key:"_getAdOptions",value:function(e){var t={},a=this._playerManager.getBreakManager().getBreakById(e.breakId);if(a){var n=this._playerManager.getBreakManager().getBreakClipById(e.breakClipId);t.url=n.contentId,t.contentType=n.contentType,t.title=n.title,t.position=a.breakClipIds.indexOf(n.id)+1,t.duration=n.duration,t.clickThroughUrl=n.clickThroughUrl,t.posterUrl=n.posterUrl,t.skipOffset=n.whenSkippable,t.linear=!0}return t}},{key:"_sendEventAndCustomMessage",value:function(e,t){this._logger.debug(e.toUpperCase(),t),this._player.dispatchEvent(new g(e,t)),this._context.sendCustomMessage(o.CUSTOM_CHANNEL,void 0,new y(e,t))}}]),e}();t.ReceiverAdsManager=v}])});
```
Comment (OrenMe) [inline dist/playkit-cast-receiver.js:1]: remove prettier/eslint rules from minified and source map

Comment (dan-ziv) [reply dist/playkit-cast-receiver.js:1]: done

File: .prettierignore [hunk 3]
```diff
@@ -1,5 +1,6 @@
-/dist
-yarn.lock
```
Comment (Yuvalke) [inline .prettierignore:2]: why did you change the yarn.lock?

File: .prettierignore [hunk 4]
```diff
@@ -1,5 +1,6 @@
...
+dist
+yarn.lock1
```
Comment (Yuvalke) [inline .prettierignore:2]: why did you change to yarn lock1?

Comment (dan-ziv) [reply .prettierignore]: typo
